### PR TITLE
Added events in CacheResponse middleware + tests

### PR DESCRIPTION
**Pull request by AlexVanderbist**

### CODE REVIEW

File: src/Events/ServedCachedResponse.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Spatie\ResponseCache\Events;
+
+use Illuminate\Http\Request;
+
+class ServedCachedResponse
```
Comment (freekmurze) [inline src/Events/ServedCachedResponse.php]: Rename to `CacheMissed`

File: src/Events/ServedActualResponse.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Spatie\ResponseCache\Events;
+
+use Illuminate\Http\Request;
+
+class ServedActualResponse
```
Comment (freekmurze) [inline src/Events/ServedActualResponse.php]: Rename to `ResponseCacheHit`

File: src/Events/ServedActualResponse.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+{
+    public $request;
+
+    /** @param  Request  $request */
```
Comment (sebastiandedeyne) [inline src/Events/ServedActualResponse.php]: FQCN 😈 

(fyi, can be set up in PHPStorm's prefs)

File: tests/Commands/FlushCommandTest.php [hunk 4]
```diff
@@ -21,4 +24,23 @@ public function it_will_clear_the_cache()
...
+
+    /** @test */
+    public function it_will_fire_events_when_clearing_the_cache()
+    {
+        $mock = $this->getMockBuilder('stdClass')->setMethods(['callback'])->getMock();
+
+        $mock->expects($this->exactly(2))
+            ->method('callback');
+
+        Event::listen(FlushingResponseCache::class, function () use ($mock) {
```
Comment (freekmurze) [inline tests/Commands/FlushCommandTest.php]: There are dedicated methods with which you can assert if an event was sent. For some examples take a look at the tests of laravel-uptime-monitor.

File: src/Events/ServedActualResponse.php (hunk 3 above)
Comment (AlexVanderbist) [inline src/Events/ServedActualResponse.php]: Thanks, I configured that now 😅

**Comment by freekmurze** [comment]
Could you also add these events to the readme?

**Comment by sebastiandedeyne** [comment]
Shouldn't the event names be more consistent? Either prefix them with `Response` or not.

**Comment by AlexVanderbist** [comment]
Maybe like this then?
- `ResponseCacheFlushing`
- `ResponseCacheFlushed`
- `ResponseCacheHit`
- `ResponseCacheMissed`

**Comment by sebastiandedeyne** [comment]
Sounds good to me!

**Comment by freekmurze** [comment]
👍

**Comment by freekmurze** [comment]
Thanks, I'll rename those events myself.

**Pull request merged**